#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "navit_controller"
gen = ParameterGenerator()
# params "enable_local_planning" is a bool type parameter
# bool_t is a boolean type
# 0 is the param level, 0 means global, 1 means local, 2 means default, 3 means advanced
# "Enable local planning." is the description of the parameter
# True is the default value of the parameter
# False is the minimum value of the parameter
# True is the maximum value of the parameter

gen.add("enable_local_planning", bool_t, 0, "Enable local planning.", True)
gen.add("local_plan_frequency", double_t, 0, "Local plan frequency.", 5.0, 1.0, 10.0)
gen.add("control_wait_timeout", double_t, 0, "Control wait timeout.", 0.5, 0.0, 10.0)
gen.add("blocked_timeout", double_t, 0, "blocked timeout.", 10.0, -1.0, 100.0)          #-1.0 means no timeout

exit(gen.generate(PACKAGE, PACKAGE, "NavitControllerReconfigure"))